kythe/rust/indexer/src/indexer/entries.rs

```diff
@@ -115,30 +116,43 @@ impl<'a> EntryEmitter<'a> {
     ///
     /// # Errors
     /// If an error occurs while writing the entry, an error is returned.
-    pub fn emit_diagnostic<S: Into<String>>(
+    pub fn emit_diagnostic(
         &mut self,
         source_vname: &VName,
-        diagnostic_vname: &VName,
         message: &str,
-        details: Option<S>,
-        url: Option<S>,
+        details: Option<&str>,
+        url: Option<&str>,
     ) -> Result<(), KytheError> {
+        // Combine diagnostic message fields and create sha256 sum
+        let combined_description =
+            format!("{}||{}||{}", message, details.unwrap_or("NONE"), url.unwrap_or("NONE"));
+        let mut sha256 = Sha256::new();
+        sha256.update(combined_description.as_str().as_bytes());
+        let bytes = sha256.finalize();
+        let sha256sum = hex::encode(bytes);
```

Comment:
sha256 works; maybe a bit heavyweight but we shouldn't be going down this path frequently.

Reply:
Another option is to base64 but that would lead to long signatures.

Reply:
The C++ indexer uses the base64 encoding of a SHA256 hash for all signatures longer than 43 bytes. This is fine here :-)